### PR TITLE
Add warning labels to the provisioning function in the AWS Fota sample

### PR DESCRIPTION
**Pull request by sigvartmh**
This also is needed for nRF Cloud

### CODE REVIEW

**Comment by sigvartmh** [comment]
@ru-fu Do you have time to look over the change to documentation it's also FYI due to your work on the same documentation. I haven't spell/grammar checked it yet(I'll do it a bit later today) but tell me if you think the information present is lacking or not precise enough.

**Comment by sigvartmh** [comment]
@hakonfam @Rallare

File: samples/nrf9160/aws_fota/CMakeLists.txt [hunk 1]
```diff
@@ -6,8 +6,21 @@
 
 cmake_minimum_required(VERSION 3.8.2)
 
+
```
Comment (hakonfam) [inline samples/nrf9160/aws_fota/CMakeLists.txt]:
```suggestion
```

File: samples/nrf9160/aws_fota/src/main.c [hunk 2]
```diff
@@ -268,58 +269,50 @@ static void broker_init(const char *hostname)
...
+		err = nrf_inbuilt_key_write(sec_tag, cred[type],
+				certificates[type], cert_len[type]);
 		printk("nrf_inbuilt_key_write => result=%d\n", err);
-		if (err) {
-			printk("CLOUD_CLIENT_PUBLIC_CERTIFICATE err: %d",
-				err);
-			return err;
-		}
 	}
+
```
Comment (sigvartmh) [inline samples/nrf9160/aws_fota/src/main.c]:
```suggestion

```

File: samples/nrf9160/aws_fota/Kconfig [hunk 3]
```diff
@@ -10,38 +10,50 @@ config APP_VERSION
...
 	default "v1.0.0"
 
+config USE_NRF_CLOUD
+	bool "Use nRF Cloud certificates"
+	default n
+
+if !USE_NRF_CLOUD
 config CLOUD_CERT_SEC_TAG
-	int "Secure tag fo TLS credentials"
+	int "Secure tag for TLS credentials"
```
Comment (ru-fu) [inline samples/nrf9160/aws_fota/Kconfig]: In LTE Link Monitor, this is called security tag, not secure tag. Which is correct? We need to be consistent in terminology.

**Comment by ru-fu** [comment]
Please check my updates and squash if okay.

File: samples/nrf9160/aws_fota/Kconfig (hunk 3 above)
Comment (sigvartmh) [reply samples/nrf9160/aws_fota/Kconfig]: Seems we should call it security tag that's what it's called in the zephyr documentation my bad.

Comment (ru-fu) [reply samples/nrf9160/aws_fota/Kconfig]: Should we change it here or in the PR for updating the doc?

Comment (sigvartmh) [reply samples/nrf9160/aws_fota/Kconfig]: I'll add it here.

File: samples/nrf9160/aws_fota/Kconfig [hunk 4]
```diff
@@ -10,38 +10,50 @@ config APP_VERSION
...
+	default n
 	help
-	  Default is set to be the NRF Cloud MQTT broker
-
-config MQTT_BROKER_PORT
-	int "AWS IoT MQTT broker port"
-	default 8883
+	  If enabled, the sample provisions server certificates into
+	  the modem by storing the certificates defined in the
+	  certificates.h file in the modem under the given secure tag.
```
Comment (ru-fu) [inline samples/nrf9160/aws_fota/Kconfig]: security tag

File: samples/nrf9160/aws_fota/README.rst [hunk 5]
```diff
@@ -61,9 +61,15 @@ Also, make sure to download one of the `root CA certificates <https://docs.aws.a
 
 Then, attach the policy you created earlier to the certificates. You must also add the information from the Root CA, Public Certificate, and Private Key into :file:`certificates.h`.
 
-When these steps are done, deselect ``CONFIG_USE_PROVISIONED_CERTIFICATES`` in menuconfig.
-This will write your new certificates to the secure tag :c:type:`sec_tag_t` defined in menuconfig ``CONFIG_CLOUD_CERT_SEC_TAG`` into the modem.
+When these steps are done, select ``CONFIG_PROVISION_CERTIFICATES`` in Kconfig.
+This will write your new certificates into the modem, to the secure tag :c:type:`sec_tag_t` defined in the Kconfig variable ``CONFIG_CLOUD_CERT_SEC_TAG``.
```
Comment (ru-fu) [inline samples/nrf9160/aws_fota/README.rst]: security tag

File: samples/nrf9160/aws_fota/Kconfig (hunk 4 above)
Comment (sigvartmh) [reply samples/nrf9160/aws_fota/Kconfig]: fixed.

File: samples/nrf9160/aws_fota/README.rst (hunk 5 above)
Comment (sigvartmh) [reply samples/nrf9160/aws_fota/README.rst]: fixed.

File: samples/nrf9160/aws_fota/src/main.c [hunk 6]
```diff
@@ -329,10 +321,11 @@ static int client_id_get(char *id_buf)
 #if !defined(CONFIG_CLOUD_CLIENT_ID)
 	enum at_cmd_state at_state;
 	char imei_buf[IMEI_LEN + 5];
-
 	int err = at_cmd_write("AT+CGSN", imei_buf, (IMEI_LEN + 5), &at_state);
+
```
Comment (sigvartmh) [inline samples/nrf9160/aws_fota/src/main.c]: intentional.

**Comment by carlescufi** [comment]
@lemrey @rlubos can we merge this?

**Comment by rlubos** [comment]
@sigvartmh We've had to revert this PR as it broke the CI, please resubmit it.

**Pull request merged**